### PR TITLE
[IBM mq] support for mTLS

### PR DESCRIPTION
**Pull request by paologallinaharbur**
This PR introduces some changes, in particular:
 - adds support for mTLS
 - adds a e2e tests for mTLS
 - updates the outdated image for the standard e2e test
 - https://github.com/newrelic/docs-website/pull/9365 Docs
 
 During the review please focus on `exporters/ibmmq/ibmmq.json.tmpl` and `exporters/ibmmq/ibmmq-config.yml.sample`

### CODE REVIEW

File: exporters/ibmmq/ibmmq-config.yml.sample [hunk 1]
```diff
@@ -32,4 +32,20 @@ integrations:
...
+        # https://www.ibm.com/docs/en/ibm-mq/9.1?topic=variables-mqsslkeyr
+        # mqsslkeyr: /key
+
+        # Path to the ccdt file. Two different environment variable will be set automatically: MQCCDTURL and IBMMQ_CONNECTION_CCDTURL.
+        # MQCCDTURL is needed by the IBMMQ client, IBMMQ_CONNECTION_CCDTURL by the exporter.
+        # ccdt_url: file:///ccdt.json
+
+        # IBMMQ Client needs permissions to write in a directory specified in the HOME environment variable
+        # https://www.ibm.com/support/pages/apar/IT32860
+        # home: /tmp
```
Comment (paologallinaharbur) [inline exporters/ibmmq/ibmmq-config.yml.sample:47]: should we change this variable name by adding a prefix?

File: exporters/ibmmq/e2e/entrypoint-consumer.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+while true; do
+    echo ${MQ_APP_PASSWORD} | /opt/mqm/samp/bin/amqsgetc DEV.QUEUE.1 QM1
```
Comment (kang-makes) [inline exporters/ibmmq/e2e/entrypoint-consumer.sh:4]: Do not we need to add a `sleep` here? I think this loop is going to spam the MQ and every core available.

Comment (kang-makes) [reply exporters/ibmmq/e2e/entrypoint-consumer.sh:4]: Oh, it is a consumer, is blocking. Nevermind.

File: exporters/ibmmq/ibmmq-config.yml.sample (hunk 1 above)
Comment (kang-makes) [reply exporters/ibmmq/ibmmq-config.yml.sample:47]: No. It was mandatory that wa: https://www.ibm.com/docs/en/ibm-mq/9.1?topic=reference-environment-variables

File: exporters/ibmmq/exporter.yml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # name of the exporter, should match with the folder name
 name: ibmmq
 # version of the package created
-version: 0.0.3
+version: 0.0.4
```
Comment (kang-makes) [inline exporters/ibmmq/exporter.yml:4]: feature?

**Review by kang-makes — APPROVED** [review]

File: exporters/ibmmq/ibmmq.json.tmpl [hunk 4]
```diff
@@ -56,5 +56,27 @@
...
+    {{ if (or .config.mqsslkeyr .config.ccdt_url .config.ld_library_path .config.home) }}
+    "env": {
+            {{ if .config.mqsslkeyr}}
+            "MQSSLKEYR":"{{.config.mqsslkeyr}}",
+            {{end}}
+
+            {{ if .config.ccdt_url }}
+            "MQCCDTURL": "{{.config.ccdt_url}}",
+            "IBMMQ_CONNECTION_CCDTURL": "{{.config.ccdt_url}}",
+            {{end}}
```
[the remark this answers is not in the document]
Comment (gsanchezgavier) [reply exporters/ibmmq/ibmmq.json.tmpl:70]: is ok, if needed in the feature we could expose both without breaking the config i think

**Review by gsanchezgavier — APPROVED** [review]
thanks!

File: exporters/ibmmq/ibmmq.json.tmpl (hunk 4 above)
[the remark this answers is not in the document]
Comment (paologallinaharbur) [reply exporters/ibmmq/ibmmq.json.tmpl:70]: yes, I though the same 😄

File: exporters/ibmmq/exporter.yml (hunk 3 above)
Comment (paologallinaharbur) [reply exporters/ibmmq/exporter.yml:4]: mmmm, yes!

**Pull request merged**